features/cellular/framework/AT/AT_CellularContext.cpp

```diff
@@ -0,0 +1,871 @@
+/*
+ * Copyright (c) 2018, Arm Limited and affiliates.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "AT_CellularContext.h"
+#include "AT_CellularNetwork.h"
+#include "AT_CellularStack.h"
+#include "CellularLog.h"
+#include "CellularUtil.h"
+#include "CellularSIM.h"
+#include "UARTSerial.h"
+#include "nsapi_ppp.h"
+#include "nsapi_dns.h"
+
+using namespace mbed_cellular_util;
+using namespace mbed;
+
+AT_CellularContext::AT_CellularContext(ATHandler &at, CellularDevice *device, const char *apn, nsapi_ip_stack_t stack) :
+        AT_CellularBase(at), _ip_stack_type_requested(DEFAULT_STACK), _is_connected(false), _is_blocking(true),
```

Comment:
_ip_stack_type_requested(DEFAULT_STACK) -> _ip_stack_type_requested(stack) ?

Reply:
true, will fix

Reply:
Can we really make API to prefer some IP version? Could just use a flag to prefer non-IP and/or control plane?